packages/doc-gen/src/index.ts

```diff
@@ -0,0 +1,145 @@
+import {
+  compile,
+  compilerAssert,
+  Decorator,
+  DocContent,
+  DocParamTagNode,
+  Enum,
+  getTypeName,
+  navigateProgram,
+  NodeHost,
+  Model,
+  Operation,
+  Program,
+  Projection,
+  IdentifierNode,
+  StringLiteral,
+  SyntaxKind,
+  Type,
+  TypeReferenceNode
+} from "@cadl-lang/compiler";
+import * as fs from 'fs';
+
+export async function generateDocs(mainfile: string) {
```

Comment:
nit: @timotheeguerin, should the entry point param be a single main file or a folder?

Reply:
We can rename that to `main` the `compile` function of the compiler just accept a path to a file or directory.